### DISCUSSION TITLE
useQuery hook not updated after parameter changes

### DISCUSSION THREAD

**Discussion by mickael-h** [TanStack/query#2347] [Q&A] [answered]
I made a query for a piece of data that is constantly being updated server-side, and so it refreshes periodically.
``` 
  const [user, setUser] = useRecoilState(userState);
  const {status, data} = useQuery('points', () => getPoints(user.uuid), {
    refetchInterval: 5000,
  });
```
When `user.uuid` is updated, the Recoil state re-renders the component, but the useQuery hook is not updated and still uses the old `uuid`. How do I make the useQuery hook update the function when it changes?
![image](https://user-images.githubusercontent.com/7953005/120922460-8d859480-c6c9-11eb-94b6-437af272a3c3.png)
![image](https://user-images.githubusercontent.com/7953005/120922450-852d5980-c6c9-11eb-9c9f-bc07a5bf3e60.png)

**Comment by TkDodo** [comment, marked as answer]
react-query will _not_ refetch on every re-render, that would be waaay too much. It will refetch if the key changes, so please put all your "dependencies" for the queryFn into the query key. Your queryFn uses `user.uuid`, so if you want react-query to refetch whenever the id changes, you have to make it part of the queryKey:

```
  const [user, setUser] = useRecoilState(userState);
  const {status, data} = useQuery(['points', user.uuid], () => getPoints(user.uuid), {
    refetchInterval: 5000,
  });
```

**Reply by mickael-h** [reply]
That worked. Thanks for the help 👍

**Reply by shuvo-me** [reply]
@TkDodo thanks for your help.
